### DISCUSSION TITLE
TypeError: data type '>u16' not understood

### DISCUSSION THREAD

**Discussion by nischayn99** [CCSDSPy/ccsdspy#93] [General]
Hi @ddasilva ,

I'm getting this error 'TypeError: data type '>u16' not understood' when I try to parse for one Packet Structure. I've attached the code and testfile for your reference. Please change the filename in Line 6 accordingly.

Thanks,
Nischay 
[MISE TEST.zip](https://github.com/CCSDSPy/ccsdspy/files/12431191/MISE.TEST.zip)

**Comment by ddasilva** [comment]
Hi @nischayn99,

Your "Time" field is defined as a 128 bit unsigned, which would require a 16-byte integer to represent. There is no native type in x86/x86_64 large enough to support this, which is why you are seeing it fail to respond to the '>u16' data type.

What I recommend you do is decode this field as an array of bytes, and then use a [custom converter](https://docs.ccsdspy.org/en/latest/user-guide/converters.html#creating-user-defined-transformations) to convert them to an instance of a class that supports unlimited length integers. I believe in Python 3.x the `int` type can do this, and therefore your converter could return an array of `dtype=object` with each element being a reference to an `int` object.

Some more info on unlimited length integers in python:
https://stackoverflow.com/questions/538551/handling-very-large-numbers-in-python

**Comment by nischayn99** [comment]
Thank you, @ddasilva
